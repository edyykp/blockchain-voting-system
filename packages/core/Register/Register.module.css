.container {
  background-color: var(--gray-700);
  min-height: 65vh;
  min-width: 50vw;
  border-radius: 50px;
  -webkit-box-shadow: 2px 2px 20px 3px #333333;
  box-shadow: 2px 2px 20px 3px #333333;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding-left: 14rem;
  padding-right: 14rem;
  width: 50%;
  position: absolute;
  left: 25%;
  margin-top: -300px;
  text-align: center;
}

.errorWrapper {
  width: 100%;
  border: 4px solid var(--primary);
  background-color: darksalmon;
  text-align: center;
  color: var(--primary);
  padding-block: 5px;
  font-size: 1.2rem;
  font-weight: 600;
  border-radius: 5px;
}

.wrapper {
  width: 100%;
  text-align: center;
}

.title {
  color: var(--white);
  font-size: 40px;
}

.subtitle {
  font-size: 30px;
  color: var(--gray-900);
}

.text {
  color: var(--gray-900);
}

.link {
  cursor: pointer;
  font-weight: 500;
  color: var(--primary);
}

.link:hover {
  border-bottom: 1px solid var(--primary);
}

@media screen and (max-width: 1440px) {
  .container {
    padding-left: 7rem;
    padding-right: 7rem;
    left: 25%;
  }

  .title {
    font-size: 35px;
  }

  .subtitle {
    font-size: 25px;
  }
}

@media screen and (max-width: 768px) {
  .container {
    padding-left: 5rem;
    padding-right: 5rem;
    min-width: 70vw;
    width: 70%;
    left: 5%;
  }

  .title {
    font-size: 30px;
  }

  .subtitle {
    font-size: 20px;
  }
}

@media screen and (max-width: 480px) {
  .container {
    min-width: 90vw;
    padding-left: 2rem;
    padding-right: 2rem;
    width: 90%;
  }

  .title {
    font-size: 30px;
  }

  .subtitle {
    font-size: 20px;
  }
}
