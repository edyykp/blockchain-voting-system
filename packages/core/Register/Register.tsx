import { Button, Input } from '@packages/components';
import { useSiteProperties, useAuthContext } from '@packages/config';

import styles from './Register.module.css';

export const Register = () => {
  const auth = useAuthContext();
  const valueOf = useSiteProperties();

  const text = {
    title: valueOf('register_form_title'),
    subtitle: valueOf('register_form_subtitle'),
    footer: valueOf('register_form_footer'),
    signin: valueOf('signin'),
  };

  return (
    <form className={styles.container} data-testid="register" id="register">
      <div className={styles.wrapper}>
        <h1 className={styles.title}>{text.title}</h1>
        <h2 className={styles.subtitle}>{text.subtitle}</h2>
      </div>
      <div className={styles.wrapper}>
        <Input type="text" field="email" isRequired={true} />
        <Input type="password" field="password" isRequired={true} />
        <Input type="password" field="confirm password" isRequired={true} />
      </div>
      <Button theme="primary" size="lg" buttonText="Register" />
      <p className={styles.text}>
        {text.footer}
        <a className={styles.link} onClick={auth.changeText}>
          {text.signin}
        </a>
      </p>
    </form>
  );
};
