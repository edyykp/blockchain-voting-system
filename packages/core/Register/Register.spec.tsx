import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Register } from './Register';

const setup = () => {
  return {
    ...render(<Register />),
  };
};

describe('Registerâ†’', () => {
  it('renders', () => {
    const { container, getByTestId } = setup();

    expect(getByTestId('register')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
