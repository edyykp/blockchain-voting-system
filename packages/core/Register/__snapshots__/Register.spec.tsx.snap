// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registerâ†’ renders 1`] = `
<div>
  <form
    class="container"
    data-testid="register"
    id="register"
  >
    <div
      class="wrapper"
    >
      <h1
        class="title"
      >
        Vote your favourite Formula 1 driver
      </h1>
      <h2
        class="subtitle"
      >
        Register here
      </h2>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="container"
      >
        <input
          autocomplete="off"
          class="input"
          placeholder="Enter your email"
          required=""
          type="text"
        />
        <label
          class="label "
          for="email"
        >
          Email
        </label>
      </div>
      <div
        class="container"
      >
        <input
          autocomplete="off"
          class="input"
          placeholder="Enter your password"
          required=""
          type="password"
        />
        <label
          class="label "
          for="password"
        >
          Password
        </label>
      </div>
      <div
        class="container"
      >
        <input
          autocomplete="off"
          class="input"
          placeholder="Enter your password"
          required=""
          type="password"
        />
        <label
          class="label "
          for="confirm password"
        >
          Confirm password
        </label>
      </div>
    </div>
    <a
      class="button  primary
      false
      false
      largeButton
      "
    >
      Register
    </a>
    <p
      class="text"
    >
      Already have an account? 
      <a
        class="link"
      >
        Sign in
      </a>
    </p>
  </form>
</div>
`;
